### PR TITLE
fix: Fix issue #3244

### LINKED ISSUE #3244: [BUG] Transaction UI doesn't fit for long category names
### Please confirm the following

- [X] I checked [the current issues](https://github.com/Ivy-Apps/ivy-wallet/issues) for duplicate problems

### Describe the bug

In Transaction log when a longer account name goes shrunk and account name not displayed fully

### To Reproduce

1. 

### Expected behavior

When a longer account name comes it should flow on second line

### Screenshots

Screenshot_2024-06-14-07-17-01-26_5e9f1c73c4ea0ad7b9937a61f977427e

### App version

4.6.3  Google Playstore

### Smartphone

OnePlus Nord CE 2 5G 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by giuseppe2028**
## Pull Request (PR) Checklist
Please check if your pull request fulfills the following requirements:
- [x] The PR is submitted to the `main` branch.
- [x] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the rules.
- [x] I've read and understand the [Developer Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Guidelines.md).
- [x] I confirm that I've run the code locally and everything works as expected.
- [x] 🎬 I've attached a **screen recoding** of the changes. 

> Tip: drag & drop the video to the PR description.

## What's changed?

Describe with a few bullets **what's new:**
- Fix issue for tag too long
- Add a little padding for a tag, in order to be visible even if the message is to long

> 💡 Tip: Please, attach screenshots and screen recordings. It helps a lot!

## Risk Factors

**What may go wrong if we merge your PR?**

- a
- b

**In what cases your code won't work?**

https://github.com/Ivy-Apps/ivy-wallet/assets/78317934/2d184ff9-c3ac-4117-9b2b-32c0c6561e01

- a
- b

## Does this PR closes any GitHub Issues?

Check **[Ivy Wallet Issues](https://github.com/Ivy-Apps/ivy-wallet/issues)**.

- Closes #3244 

> Replace `ISSUE_NUMBER` with the number of the GitHub issue that this PR is fixing. If you've done that correctly, you'll see the issue title linked when previewing your PR description.

> **!Note:** Do **not** link the PR number. Link the number/id of the GitHub Issue from [issues](https://github.com/Ivy-Apps/ivy-wallet/issues).

## Troubleshooting CI failures ❌

GitHub Actions failing? Read our [CI Troubleshooting guide](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/CI-Troubleshooting.md).

### CODE REVIEW

**Comment by ILIYANGERMANOV** [comment]
@giuseppe2028 you need to update the Paparazzi screenshots

https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/CI-Troubleshooting.md#paparazzi-tests

File: temp/legacy-code/src/main/java/com/ivy/legacy/ui/component/transaction/TransactionCard.kt [hunk 1]
```diff
@@ -397,7 +401,7 @@ private fun TransactionBadge(
             .background(backgroundColor, UI.shapes.rFull)
             .clickable {
                 onClick()
-            },
+            }.padding(end = 10.dp),
```
Comment (ILIYANGERMANOV) [inline temp/legacy-code/src/main/java/com/ivy/legacy/ui/component/transaction/TransactionCard.kt:404]: Remove this padding and increase the last spacer from `20.dp` to some more. Also why do we add more padding? What's the benefit?

File: temp/legacy-code/src/main/java/com/ivy/legacy/ui/component/transaction/TransactionCard.kt [hunk 2]
```diff
@@ -310,13 +314,13 @@ private fun TransactionHeaderRow(
             )
         }
     } else {
-        Row(
+        FlowRow(
+            horizontalArrangement = Arrangement.spacedBy(12.dp),
             modifier = Modifier.padding(horizontal = 20.dp),
-            verticalAlignment = Alignment.CenterVertically
+            verticalArrangement = Arrangement.spacedBy(8.dp)
         ) {
```
Comment (ILIYANGERMANOV) [inline temp/legacy-code/src/main/java/com/ivy/legacy/ui/component/transaction/TransactionCard.kt:321]: The FlowRow looks good, how the items rendee when they are on a single line (most cases)? They must be vertically centered.

When you update the Paparazzi screenshots and commit them, we'll see that case hopefully.

Comment (ILIYANGERMANOV) [reply temp/legacy-code/src/main/java/com/ivy/legacy/ui/component/transaction/TransactionCard.kt:321]: Which reminds, please add a new Paparazzi test with the account going on a separate row

File: temp/legacy-code/src/main/java/com/ivy/legacy/ui/component/transaction/TransactionCard.kt (hunk 1 above)
Comment (giuseppe2028) [reply temp/legacy-code/src/main/java/com/ivy/legacy/ui/component/transaction/TransactionCard.kt:404]: The purpose of this padding is to ensure that the text within the blue transaction badge does not extend beyond its borders.
Screenshot 2024-06-20 alle 22 42 02 
Unfortunately your suggestion doesn't work. Instead with the padding the result is: 
Screenshot 2024-06-20 alle 22 46 27

Comment (ILIYANGERMANOV) [reply temp/legacy-code/src/main/java/com/ivy/legacy/ui/component/transaction/TransactionCard.kt:404]: Good, makes sense! 💯

File: temp/legacy-code/src/main/java/com/ivy/legacy/ui/component/transaction/TransactionCard.kt (hunk 2 above)
Comment (giuseppe2028) [reply temp/legacy-code/src/main/java/com/ivy/legacy/ui/component/transaction/TransactionCard.kt:321]: ok i understand why paparazzi tests not passed, it is due to something that I've never modified. Now, I have to follow the guide in "https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/CI-Troubleshooting.md#paparazzi-tests" and update each single screen?
Screenshot 2024-06-21 alle 00 48 43
This is an example of a UI that has been modified

Comment (ILIYANGERMANOV) [reply temp/legacy-code/src/main/java/com/ivy/legacy/ui/component/transaction/TransactionCard.kt:321]: @giuseppe2028 you modified the transaction and it's used on many screens. You can run:
```
./gradlew recPaD
```

to update all screens that have changed at once

**Comment by giuseppe2028** [comment]
I have tried all the possible solution, but the test does not run yet.

**Comment by ILIYANGERMANOV** [comment]
> I have tried all the possible solution, but the test does not run yet.

@giuseppe2028 try increasing this to `0.005`

https://github.com/Ivy-Apps/ivy-wallet/blob/8f0360f02542ea381243601a2dce635415943f2d/shared/ui/testing/src/main/java/com/ivy/ui/testing/PaparazziScreenshotTest.kt#L14

**Comment by ILIYANGERMANOV** [comment]
We recently updated Compose and there might be flaky Paparazzi tests. Also update your branch to be on top of the latest `main`. Make sure to commit the screenshots for all Paparazzi.

**Comment by ILIYANGERMANOV** [comment]
This https://github.com/Ivy-Apps/ivy-wallet/issues/3243 should improve this suboptimal developer experience

File: shared/ui/testing/src/main/java/com/ivy/ui/testing/PaparazziScreenshotTest.kt [hunk 3]
```diff
@@ -4,9 +4,17 @@ import androidx.compose.runtime.Composable
 import app.cash.paparazzi.DeviceConfig
 import app.cash.paparazzi.Paparazzi
 import com.ivy.design.system.IvyMaterial3Theme
+import org.junit.Before
 import org.junit.Rule
+import java.util.Locale
 
 open class PaparazziScreenshotTest {
+
+    @Before
```
Comment (giuseppe2028) [inline shared/ui/testing/src/main/java/com/ivy/ui/testing/PaparazziScreenshotTest.kt:13]: @ILIYANGERMANOV I added this snippet to ensure that paparazzi doesn’t rely on local settings, which were causing issues with screen registration. This should standardize the tests and prevent any local settings from affecting them.

Comment (ILIYANGERMANOV) [reply shared/ui/testing/src/main/java/com/ivy/ui/testing/PaparazziScreenshotTest.kt:13]: Perfect 💯

**Review by ILIYANGERMANOV — APPROVED** [review]

**Pull request merged**